.github/workflows/public-upgrade.yml

```
@@ -0,0 +1,220 @@
+      HQBASE_E2E_REPO_CONNECTION_UUID: ${{ vars.HQBASE_E2E_REPO_CONNECTION_UUID }}
+      HQBASE_E2E_BUILD_TOKEN_UUID: ${{ vars.HQBASE_E2E_BUILD_TOKEN_UUID }}
+      HQBASE_STAGING_SENDER: hello@${{ secrets.HQBASE_E2E_EMAIL_DOMAIN }}
+      HQBASE_STAGING_URL: https://${{ secrets.HQBASE_E2E_APP_HOSTNAME }}
+      TEST_CANDIDATE_VERSION: ${{ inputs.candidate_version }}
+      TEST_SUCCESSOR_VERSION: ${{ inputs.successor_version }}
+      UPGRADE_EDGE: ${{ matrix.edge }}
+      HQBASE_STAGING_PUBLIC_UPGRADE: "1"
+      HQBASE_E2E_UPDATE_API_TOKEN: ${{ secrets.HQBASE_E2E_UPDATE_API_TOKEN }}
+      DEPLOYMENT_NAME: public-${{ github.run_id }}-${{ matrix.edge }}
+      STAGING_WORKER_NAME: hqbase-public-${{ github.run_id }}-${{ matrix.edge }}
+      HQBASE_UPGRADE_RECEIPT: ${{ runner.temp }}/upgrade-${{ matrix.edge }}.json
```

actionlint (1.7.12)

[error] context "runner" is not allowed here. available contexts are "github", "inputs", "matrix", "needs", "secrets", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)